### ISSUE TITLE
Can't run our suite of tests in parallel

### ISSUE THREAD

**Issue by alb-i986** [junit-team/junit4#1297]
```
mvn clean test -Pfast-tests
```

This will run our unit tests in parallel (2 classes). But it doesn't work.
Depending on how the test classes are scheduled, I get either a BUILD FAILURE (see e.g. https://travis-ci.org/alb-i986/junit/jobs/129814767)

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.18.1:test (default-test) on project junit: Execution default-test of goal org.apache.maven.plugins:maven-surefire-plugin:2.18.1:test failed: The forked VM terminated without properly saying goodbye. VM crash or System.exit called?
```

or the command hangs, and I need to CTRL-C.

The former is probably due to [SystemExitTest](https://github.com/junit-team/junit4/blob/master/src/test/java/org/junit/tests/running/core/SystemExitTest.java), [JUnitCoreReturnsCorrectExitCodeTest](https://github.com/junit-team/junit4/blob/master/src/test/java/org/junit/tests/running/core/JUnitCoreReturnsCorrectExitCodeTest.java), na possibly others.
The latter must be due to some nasty concurrency issue.

Since tracking down concurrency issues is not an easy one, I would suggest to simply remove the profile fast-tests.

**Comment by marcphilipp** [comment]
+1 I wasn't a fan of introducing it in the first place.

**Comment by kcooney** [comment]
This should be an easy fix; just revert 9d2817c
